### PR TITLE
[Synthetics] Docs on secrets and params

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes https://github.com/elastic/observability-docs/issues/805

Adds docs for params and secrets.

### CODE REVIEW

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  }
+  return {
+    params: {
+      url,
+    },
+  };
+};
+----
+
+Note that in the above example we use the `env` variable, which corresponds to the value of the `-e` or `--environment` CLI
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-params-secrets.asciidoc:76]: Can we also say users can also rely on NODE_ENV for defining environments as we can deprecate this flag in future versions.

**Review by vigneshshanmugam — COMMENTED** [review]
Great stuff @andrewvc 🎉 Added some suggestions, but looks good overall

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+[discrete]
+[[synthetics-configs]]
+=== Using CLI arguments to set params
+
+Use the `--suite-params` or `-s` arguments to the `elastic-synthetics` program to set parameters when running `elastic-synthetics` on
+the command line. The provided map will be merged over any existing variables defined in the `synthetics.config.{js,ts}` file.
+
+To override the `url` parameter, you would run: `elastic-synthetics . --suite-params '{"url": "http://localhost:8080"}'`
+[discrete]
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:87]: For the docs build to pass, this [discrete] needs to be moved above the other title attribute on line 89.

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+[[synthetics-configs]]
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:89]: This title attribute is the same as the one on line 80. For the docs build to pass it needs to be unique. Maybe [[synthetics-heartbeat-options]]

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+Note that in the above example we use the `env` variable, which corresponds to the value of the `-e` or `--environment` CLI
+flag for `elastic-synthetics`, or the `environment` parameter in the `browser` monitor definition. 
+
+[discrete]
+[[synthetics-configs]]
+=== Using CLI arguments to set params
+
+Use the `--suite-params` or `-s` arguments to the `elastic-synthetics` program to set parameters when running `elastic-synthetics` on
+the command line. The provided map will be merged over any existing variables defined in the `synthetics.config.{js,ts}` file.
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:84]:
```suggestion
To set parameters when running `elastic-synthetics` on the command line, use the `--suite-params` or `-s` arguments to the `elastic-synthetics` program. The provided map is merged over any existing variables defined in the `synthetics.config.{js,ts}` file.
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+Another scenario might be that you want to use a different URL when running the tests under heartbeat then running them locally using the synthetics agent. 
+Solving these problems is where params come in. Params are variables that can be used within a synthetic suite. 
+They can be defined either in the suites config file, on the synthetics agent command line, or in a heartbeat yaml config.
+
+It should also be noted that since synthetics is a full javascript environment it is also possible to use regular environment variables via
+the standard node `process.env` global object.
+
+[discrete]
+[[synthetics-basic-params]]
+=== Parameter Basics
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:23]:
```suggestion
== Parameter Basics
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+* Specifying the parameter in the heartbeat yaml config using the `params` option.
+
+A common pattern is to use a configuration file for default values to be configured via a configuration file, using the 
+`--suite-params` CLI argument, or `params` heartbeat option to override these defaults in the given environment. 
+
+These options are discussed in detail in the sections below.
+
+[discrete]
+[[synthetics-configs]]
+=== Using a config file to set params
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:56]:
```suggestion
== Using a config file to set params
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+  };
+};
+----
+
+Note that in the above example we use the `env` variable, which corresponds to the value of the `-e` or `--environment` CLI
+flag for `elastic-synthetics`, or the `environment` parameter in the `browser` monitor definition. 
+
+[discrete]
+[[synthetics-configs]]
+=== Using CLI arguments to set params
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:81]:
```suggestion
== Using CLI arguments to set params
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+Use the `--suite-params` or `-s` arguments to the `elastic-synthetics` program to set parameters when running `elastic-synthetics` on
+the command line. The provided map will be merged over any existing variables defined in the `synthetics.config.{js,ts}` file.
+
+To override the `url` parameter, you would run: `elastic-synthetics . --suite-params '{"url": "http://localhost:8080"}'`
+[discrete]
+
+[[synthetics-configs]]
+=== Using Heartbeat options to set params
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:90]:
```suggestion
== Using Heartbeat options to set params
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+  params:
+    url: "https://elastic.github.io/synthetics-demo/"
+  source:
+    zip_url:
+      url: "https://github.com/elastic/synthetics-demo/archive/refs/heads/main.zip"
+      folder: "synthetics-tests"
+----
+
+[[synthetics-configs]]
+=== Working with secrets and sensitive values
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:112]:
```suggestion
== Working with secrets and sensitive values
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+  tags: todos-app
+  params:
+    url: "https://elastic.github.io/synthetics-demo/"
+  source:
+    zip_url:
+      url: "https://github.com/elastic/synthetics-demo/archive/refs/heads/main.zip"
+      folder: "synthetics-tests"
+----
+
+[[synthetics-configs]]
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:111]: This title attribute is the same as the one on line 80. For the docs build to pass, it needs to be unique. Maybe [[synthetics-sensitive-values]]. You also need to add an [discrete] attribute above it.

File: docs/en/observability/synthetics-params-secrets.asciidoc (hunk 1 above)
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:76]:
```suggestion
Note that we use the `env` variable in the above example, which corresponds to the value of the `-e` or `--environment` CLI
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+To override the `url` parameter, you would run: `elastic-synthetics . --suite-params '{"url": "http://localhost:8080"}'`
+[discrete]
+
+[[synthetics-configs]]
+=== Using Heartbeat options to set params
+
+When running via Heartbeat use the `params` option to set additional parameters. These will be passed through the `--suite-params` flag
+mentioned above, and have their values merged over any default values. In the example below we run the todos app, overriding the `url`
+parameter.
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:94]:
```suggestion
When running via Heartbeat use the `params` option to set additional parameters, passed through the `--suite-params` flag
mentioned above and have their values merged over any default values. In the example below we run the todos app, overriding the `url`
parameter.
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+      url: "https://github.com/elastic/synthetics-demo/archive/refs/heads/main.zip"
+      folder: "synthetics-tests"
+----
+
+[[synthetics-configs]]
+=== Working with secrets and sensitive values
+
+Your synthetics scripts may require the use of passwords or other sensitive secrets that are not known till runtime. Before we continue, it is 
+important to remember that since synthetics scripts have no limitations, a malicious script author could write a synthetics journey that 
+exfiltrates `params` and other data at runtime. It is generally best not to use truly sensitive passwords (e.g. an admin password to test an admin
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:116]:
```suggestion
exfiltrates `params` and other data at runtime. Therefore, it is generally best not to use truly sensitive passwords (e.g. an admin password to test an admin
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+panel, or a real credit card) in *any* synthetics tools. Instead, setup limited dummy accounts, or fake credit cards with limited functionality.
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:117]:
```suggestion
panel, or a real credit card) in *any* synthetics tools. Instead, set up limited demo accounts, or fake credit cards with limited functionality.
```

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+
+Use either environment variables or the heartbeat keystore to handle any values needing encryption at rest.
```
Comment (EamonnTP) [inline docs/en/observability/synthetics-params-secrets.asciidoc:119]:
```suggestion
Use either environment variables or the Heartbeat keystore to handle any values needing encryption at rest. 
```

**Review by EamonnTP — CHANGES_REQUESTED** [review]
Looks great @andrewvc Just of couple of edit suggestions, along with fixes for the docs build. Thanks!

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+  console.log(`Visiting ${params.url}`)
+})
+
+journey("My Journey", ({ page, params }) => {
+    step('launch app', async () => {
+      await page.goto(params.url);<1>
+    });
+});
+----
+<1> Note that in a typescript program you would want to instead use `params.url as string`.
```
[the remark this answers is not in the document]
Comment (andrewvc) [reply docs/en/observability/synthetics-params-secrets.asciidoc:40]: I'm targeting these docs at people who may be relatively new to TS, for instance, python devs trying to add synthetics, so anything surprising may be a good thing to add.

File: docs/en/observability/synthetics-params-secrets.asciidoc (hunk 13 above)
Comment (andrewvc) [reply docs/en/observability/synthetics-params-secrets.asciidoc:117]: Will incorporate `set up` but not the change from `dummy` -> `demo`. `demo` implies something used for demonstration, not a test.

Comment (EamonnTP) [reply docs/en/observability/synthetics-params-secrets.asciidoc]: Good point. Thanks.

**Review by EamonnTP — APPROVED** [review]

File: docs/en/observability/synthetics-params-secrets.asciidoc [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+  }
+  return {
+    params: {
+      url,
+    },
+  };
+};
+----
+
+Note that we use the `env` variable in the above example, which corresponds to the value of the `-e` or `--environment` CLI
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-params-secrets.asciidoc:72]:
```suggestion
Note that we use the `env` variable in the above example, which corresponds to the value of the `NODE_ENV` value
```

**Review by vigneshshanmugam — COMMENTED** [review]
We need to remove the `-e` flag from the docs as its removed from Synthetics CLI.

**Comment by andrewvc** [comment]
@Mergifyio backport 7.x

**Pull request merged**